{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b60101c",
   "metadata": {},
   "outputs": [],
   "source": [
    "{\n",
    " \"cells\": [\n",
    "  {\n",
    "   \"cell_type\": \"markdown\",\n",
    "   \"metadata\": {},\n",
    "   \"source\": [\n",
    "    \"# ANI pipeline con skani y fastANI\\n\",\n",
    "    \"Descarga genomas, descomprime, genera listas y calcula ANI.\\n\"\n",
    "   ]\n",
    "  },\n",
    "  {\n",
    "   \"cell_type\": \"code\",\n",
    "   \"execution_count\": null,\n",
    "   \"metadata\": {},\n",
    "   \"outputs\": [],\n",
    "   \"source\": [\n",
    "    \"!echo \\\"Ingresa taxID\\\"\\n\",\n",
    "    \"taxid = input()\\n\",\n",
    "    \"output_dir = f\\\"genomes_tax_{taxid}\\\"\\n\"\n",
    "   ]\n",
    "  },\n",
    "  {\n",
    "   \"cell_type\": \"code\",\n",
    "   \"execution_count\": null,\n",
    "   \"metadata\": {},\n",
    "   \"outputs\": [],\n",
    "   \"source\": [\n",
    "    \"!mkdir -p \\\"$output_dir\\\"\\n\",\n",
    "    \"SCINAME=$(esearch -db taxonomy -query \\\"$taxid\\\" | efetch -format xml | xtract -pattern Taxon -element ScientificName)\\n\",\n",
    "    \"if [[ -z \\\"$SCINAME\\\" ]]; then\\n\",\n",
    "    \"    print('TaxID invÃ¡lido o sin resultados')\\n\",\n",
    "    \"    exit(1)\\n\",\n",
    "    \"else:\\n\",\n",
    "    \"    print(f\\\"Organismo detectado: {SCINAME}\\\")\\n\",\n",
    "    \"!\\n\",\n",
    "    \"ncbi-genome-download bacteria \\\\\\n\",\n",
    "    \"    --genera \\\"$SCINAME\\\" \\\\\\n\",\n",
    "    \"    --assembly-level complete \\\\\\n\",\n",
    "    \"    --formats fasta \\\\\\n\",\n",
    "    \"    --section refseq \\\\\\n\",\n",
    "    \"    --output-folder \\\"$output_dir\\\" \\\\\\n\",\n",
    "    \"    --flat-output\\n\"\n",
    "   ]\n",
    "  },\n",
    "  {\n",
    "   \"cell_type\": \"code\",\n",
    "   \"execution_count\": null,\n",
    "   \"metadata\": {},\n",
    "   \"outputs\": [],\n",
    "   \"source\": [\n",
    "    \"!find \\\"$output_dir\\\" -type f -name \\\"*.fna.gz\\\" -exec gunzip -f {} \\\\;\"\n",
    "   ]\n",
    "  },\n",
    "  {\n",
    "   \"cell_type\": \"code\",\n",
    "   \"execution_count\": null,\n",
    "   \"metadata\": {},\n",
    "   \"outputs\": [],\n",
    "   \"source\": [\n",
    "    \"!find \\\"$output_dir\\\" -type f -name \\\"*.fna\\\" > genome_list.txt\\n\",\n",
    "    \"!wc -l genome_list.txt\\n\",\n",
    "    \"!head genome_list.txt\"\n",
    "   ]\n",
    "  },\n",
    "  {\n",
    "   \"cell_type\": \"code\",\n",
    "   \"execution_count\": null,\n",
    "   \"metadata\": {},\n",
    "   \"outputs\": [],\n",
    "   \"source\": [\n",
    "    \"!skani triangle -l genome_list.txt -o skani_output.tsv -t 2\"\n",
    "   ]\n",
    "  },\n",
    "  {\n",
    "   \"cell_type\": \"code\",\n",
    "   \"execution_count\": null,\n",
    "   \"metadata\": {},\n",
    "   \"outputs\": [],\n",
    "   \"source\": [\n",
    "    \"!cp genome_list.txt fastani_query_list.txt\\n\",\n",
    "    \"!cp genome_list.txt fastani_ref_list.txt\\n\",\n",
    "    \"!fastANI --ql fastani_query_list.txt --rl fastani_ref_list.txt --fragLen 3000 -t 2 -o fastani_output.tsv\"\n",
    "   ]\n",
    "  },\n",
    "  {\n",
    "   \"cell_type\": \"code\",\n",
    "   \"execution_count\": null,\n",
    "   \"metadata\": {},\n",
    "   \"outputs\": [],\n",
    "   \"source\": [\n",
    "    \"import pandas as pd\\n\",\n",
    "    \"df_skani = pd.read_csv(\\\"skani_output.tsv\\\", sep=\\\"\\\\t\\\")\\n\",\n",
    "    \"df_fastani = pd.read_csv(\\\"fastani_output.tsv\\\", sep=\\\"\\\\t\\\", header=None,\\n\",\n",
    "    \"                         names=[\\\"Query\\\", \\\"Reference\\\", \\\"ANI\\\", \\\"Fragments\\\", \\\"Total_Frags\\\"])\\n\",\n",
    "    \"print(\\\"SKANI ANI:\\\")\\n\",\n",
    "    \"display(df_skani.head())\\n\",\n",
    "    \"print(\\\"FASTANI ANI:\\\")\\n\",\n",
    "    \"display(df_fastani.head())\"\n",
    "   ]\n",
    "  }\n",
    " ],\n",
    " \"metadata\": {\n",
    "  \"kernelspec\": {\n",
    "   \"display_name\": \"Python 3\",\n",
    "   \"language\": \"python\",\n",
    "   \"name\": \"python3\"\n",
    "  },\n",
    "  \"language_info\": {\n",
    "   \"codemirror_mode\": {\n",
    "    \"name\": \"ipython\",\n",
    "    \"version\": 3\n",
    "   },\n",
    "   \"file_extension\": \".py\",\n",
    "   \"mimetype\": \"text/x-python\",\n",
    "   \"name\": \"python\",\n",
    "   \"nbconvert_exporter\": \"python\",\n",
    "   \"pygments_lexer\": \"ipython3\",\n",
    "   \"version\": \"3.10.8\"\n",
    "  }\n",
    " },\n",
    " \"nbformat\": 4,\n",
    " \"nbformat_minor\": 5\n",
    "}\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
